pkg/operator/controller/ingress/status_test.go

```diff
@@ -1580,16 +1581,62 @@ func TestComputeDNSStatus(t *testing.T) {
 				Reason: "UnsupportedEndpointPublishingStrategy",
 			}},
 		},
+		{
+			name: "DNSManaged false due to domain not matching base domain",
+			dnsConfig: &configv1.DNS{
+				Spec: configv1.DNSSpec{
+					BaseDomain:  "basedomain.com",
+					PublicZone:  &configv1.DNSZone{},
+					PrivateZone: &configv1.DNSZone{},
+				},
+			},
+			platformStatus: &configv1.PlatformStatus{
+				Type: configv1.AWSPlatformType,
+			},
+			controller: &operatorv1.IngressController{
+				Status: operatorv1.IngressControllerStatus{
+					Domain: "apps.example.com",
+					EndpointPublishingStrategy: &operatorv1.EndpointPublishingStrategy{
+						Type: operatorv1.LoadBalancerServiceStrategyType,
+						LoadBalancer: &operatorv1.LoadBalancerStrategy{
+							DNSManagementPolicy: operatorv1.UnmanagedLoadBalancerDNS,
+						},
+					},
+				},
+			},
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSManagementPolicy: iov1.UnmanagedDNS,
+				},
+			},
+			expect: []operatorv1.OperatorCondition{
+				{
+					Type:   "DNSManaged",
+					Status: operatorv1.ConditionFalse,
+					Reason: "DomainNotMatching",
+				},
+				{
+					Type:   "DNSReady",
+					Status: operatorv1.ConditionUnknown,
+					Reason: "UnmanagedDNS",
+				},
+			},
+		},
 		{
 			name: "DNSManaged false due to UnmanagedLoadBalancerDNS",
 			dnsConfig: &configv1.DNS{
 				Spec: configv1.DNSSpec{
+					BaseDomain:  "basedomain.com",
 					PublicZone:  &configv1.DNSZone{},
 					PrivateZone: &configv1.DNSZone{},
 				},
 			},
+			platformStatus: &configv1.PlatformStatus{
```

Comment:
Please keep the testcase for when the platformStatus is not set, and add a new testcase for the new case.

Reply:
This test is applicable only when `dnsManagementPolicy` is set to `Unmanaged`. The domains match in this case. I had to introduce `configv1.PlatformStatus{}` since code now always depends on `manageDNSForDomain()` which expects `PlatformStatus` to be non-nil always. 

Previously this dependency on `manageDNSForDomain()` didn't exist when `dnsManagementPolicy` is set to `Unamanaged`.

Reply:
> expects PlatformStatus to be non-nil always.
I don't think we can make that expectation - manageDNSforDomain needs to be prepared for when it is nil. Am I missing something?

Reply:
Agreed, will patch `manageDNSforDomain()` and add new tests accordingly.

Reply:
Hey @candita, what would your suggestion be for this case? 
```go
--- a/pkg/operator/controller/ingress/dns.go
+++ b/pkg/operator/controller/ingress/dns.go
@@ -208,6 +208,10 @@ func manageDNSForDomain(domain string, status *configv1.PlatformStatus, dnsConfi
                return false
        }

+       if dnsConfig == nil || status == nil {
+               return false
+       }
+
        mustContain := "." + dnsConfig.Spec.BaseDomain
        switch status.Type {
        case configv1.AWSPlatformType:
``` 
Returning `false` when either `platformStatus` or `dnsConfig` is `nil` will lead to incorrectly updating the condition to `DomainNotMatching`. And returning a `true` would also be incorrect.

There is a check for the same already ([here](https://github.com/openshift/cluster-ingress-operator/blob/d9d1a2b44cc7955a18fbedfdc973daddba67bccd/pkg/operator/controller/ingress/controller.go#L264)) but it isn't local to `manageDNSForDomain()` and thereby not covered by the tests in `status_test.go`.

Reply:
Sorry ahead of time if you already have accomplished this, but making sure we are all on the same page:

I think we need nil pointer checks before using pointers in functions, regardless if we do a nil pointer check outside of the function. We should also unit test for nil pointer exceptions. Sounds like we are talking about two different functions with possible Nil Pointer Exceptions:

**computeDNSStatus**:
Seems like the `status *configv1.PlatformStatus` is passed directly into `manageDNSForDomain`, so really it's `manageDNSForDomain`'s problem. So I think we okay here, but still worth keeping a unit test in `TestCoHope this helps and feel free to push back on some of this, but I do think this is a good opportunity to do these small maintenance tasks.mputeDNSStatus` like Candace said to make sure `computeDNSStatus` doesn't get a NIL Pointer exception in the future when code changes.

Unrelated to this PR, but`dnsConfig` this is missing a NIL Pointer check, that should probably be fixed sometime.

**manageDNSForDomain**:
I'd vote to just return `false` when given a nil pointer, since saying DNS is unmanageable is more accurate than saying`true` (it's manageable). Either way, it's much better than a nil pointer exception.

For `TestManageDNSForDomain` should have NIL Pointer test cases in it, but the test is set up a little weird, so maybe you could modify it to do something like this:
```
var status *configv1.PlatformStatus
if tc.platformType != "" {
  status = configv1.PlatformStatus{
    Type: tc.platformType,
  }

var dnsConfig *configv1.DNS
if tc.baseDomain != "" {
  dnsConfig = configv1.DNS{
    Spec: configv1.DNSSpec{
      BaseDomain: tc.baseDomain,
    },
 }
}
actual := manageDNSForDomain(tc.domain, status, dnsConfig)
```
And add the unit test cases that would trigger this `false` return value.

Reply:
I've addressed the comments.